src/engine/renderer/tr_init.cpp

```diff
@@ -77,6 +77,7 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 	cvar_t      *r_dynamicLightCastShadows;
 	cvar_t      *r_precomputedLighting;
 	Cvar::Cvar<int> r_mapOverBrightBits("r_mapOverBrightBits", "default map light color shift", Cvar::NONE, 2);
+	Cvar::Cvar<bool> r_mapClampOverBright("r_mapClampOverBright", "clamp over bright of legacy maps (enable multiplied color clamping and normalization)", Cvar::NONE, true);
```

Comment:
Why the word "map" in these cvar names? Seems backwards since these are only used when a value is NOT provided by the map. I suggest `r_defaultOverbrightBits` and `r_defaultClampOverbright` with the word "default" emphasizing that changes to their value might be ignored.

Reply:
There was originally two cvars, `r_overBrightBits` and `r_mapOverBrightBits`. The first one was configuring some hardware feature, and the engine was multiplying the lights with `r_mapOverBrightBits` and `r_overBrightBits`.

Let's imagine we had `r_mapOverBrightBits` set to 2, if there was no hardware overbright bits (`r_overBrightBits 0`), the lights were fully multiplied by 2, if there was 1 hardware overbright bit (`r_overBrightBits 1`), the lights were only multiplied by 1, and if there was 1 hardware overbright bits (`r_overBrightBits 2`), the lights were not multiplied at all.

The related `rgbGen identity` and `rgbGen identityLighting` were also related to this, as one was related to blending something with multiplied light or premultiplied light, meaning one of them had to cancel the hardware overbright multiplication, with `rgbGen identity` and `rgbGen identityLighting`  being equal if there was no hardware overbright bit (nothing to compensate)

But since our engine doesn't implement hardware overbright bit, the only cvar left is `r_mapOverBrightBits` and `rgbGen identity` and `rgbGen identityLighting` are equal (nothing to compensate).

With that being said, one may claim that to be fully compatible with hardware overbrightbits, one may only have to multiply the end result. But the life is more complicated than that. This [official Quake  3 documentation](https://icculus.org/gtkradiant/documentation/Q3AShader_Manual/ch05/pg5_1.htm) said on `6.3.1 RgbGen identityLighting` chapter:

> 6.3.1 RgbGen identityLighting
Colors will be (1.0,1.0,1.0) if running without overbright bits (NT, linux, windowed modes), or (0.5, 0.5, 0.5) if running with overbright. Overbright allows a greater color range at the expense of a loss of precision. Additive and blended stages will get this by default.

According to this documentation, the last platform to support hardware overbright bits was Windows 98 when playing fullscreen. Also, it is said in other places that the feature was not compatible with LCD screens and then required a CRT screen.

It happens that some more recent may have provided the feature more recently, as [mentionned here](https://github.com/DaemonEngine/Daemon/pull/1034#issuecomment-1933790759), but for most people the feature is lost since forever.

It means than since 20 years most mappers and developers were testing their maps and graphical effects without that feature, and may have implemented things without testing them with hardware overbright. And since we don't do hardware overbright, we have to face the ultimate limitation: the precision of the user display, which is 24-bit, 8 bit per color. One good example of thing we cannot multiply is the Wolf:ET global fog, the screen itself is clamping the multiplied global fog, and it's permitted to have very strong doubts developer expected that fog to be multiplied, it looks wrong even on non-clamped values.

Reply:
The `map` prefix in `r_mapClampOverBright` is mainly cargo cult, but it means that option sets a value that can differ per map.

For `r_mapOverBrightBits`, this is the default per-map value for when the map doesn't set `mapOverBrightBits` key in worldspawn, so I did the same for `r_mapClampOverBright`, the default per-map value for when the map doesn't set it.

Reply:
We can reword the explanation in the way `mapOverBrightBits` is a BSP worldspawn entity key, and the `r_mapOverBrightBits` cvar carries the engine default value for it when the BSP doesn't set it, hence the same name.

In the same way new `r_mapClampOverBright` engine cvar carries the default value for the new `mapClampOverBright` BSP worldspawn key. I can name those cvar and key as `r_clampOverBright` and `clampOverBright` if people prefer.  It's fine, as the 

But `mapOverBrightBits` and related cvar should be kept that way for legacy compatibility, even if the extra `map` prefix for a map key is superfluous.

Reply:
> I can name those cvar and key as r_clampOverBright and clampOverBright if people prefer. It's fine.

At least for the engine cvar, the `map` prefix reminds this value can be override by the map itself, which is not bad and may be better. If the map sets the value, it will help people understand why tweaking the engine cvar does nothing.